### DISCUSSION TITLE
SelectPicker onSearch calls when it is closed

### DISCUSSION THREAD

**Discussion by abdullaev388** [rsuite/rsuite#2710] [Q&A] [answered]
First, open this example: 
https://codesandbox.io/s/beautiful-payne-kj129b?file=/src/index.js

Then open console, then press select picker and then close it. Even though you did not search for anything, it is being called

**Comment by myNameIsDu** [comment, marked as answer]
> First, open this example: https://codesandbox.io/s/beautiful-payne-kj129b?file=/src/index.js
> 
> Then open console, then press select picker and then close it. Even though you did not search for anything, it is being called

For now, it is expected behaviour. And Is this causing you any problems?

**Reply by abdullaev388** [reply]
Why is this expected? Yes,  it is causing me some problems, i make debounced request whenever onSearch triggers and i don't want to make request when it is closed because i do it when user opens it instead.

**Reply by myNameIsDu** [reply]
Because when it is closed, the `searchKeyWord` is set to `''`, which can cause inconsistent status if not called to notify the user. 

Um, I don't completely understand your case, Maybe you need to check if it's `''`.

**Reply by abdullaev388** [reply]
No, i just need to somehow prevent its trigger when overlay is closed. Because if i add your check and user clears by himself the search text, i still need to make request with empty text

**Reply by myNameIsDu** [reply]
Okay, Thank you for the feedback.
